### PR TITLE
Feature/basic phoenix channel

### PR DESCRIPTION
**Pull request by gfodor**
This PR:

- Disables cookies + sessions in phoenix

- Adds back in phoenix channels, enabling the client to keep a persistent websocket connection open to phoenix during the session

- Sends the hub info down over the join event for the channel, and removes the no longer necessary GET api

- Adds a partitioned session_stats table and a max_occupant_count field to hubs for OKR tracking

- Adds a message events:entered that is used to update relevant info to the session_stats table when the user enters the scene

Some of the PR is due to the new elixir reformatter in 1.6, apologies.

Client PR: https://github.com/mozilla/mr-social-client/pull/161

### CODE REVIEW

File: lib/ret/peerage_provider.ex [hunk 1]
```diff
@@ -9,10 +9,11 @@ defmodule Ret.PeerageProvider do
     habitat_ip = habitat_config[:ip]
     habitat_port = habitat_config[:http_port]
```
Comment (gfodor) [inline lib/ret/peerage_provider.ex]: reformatting

File: lib/ret/hub.ex [hunk 2]
```diff
@@ -35,23 +36,31 @@ defmodule Ret.Hub do
...
+    |> validate_required([:max_occupant_count])
   end
 
   defp add_hub_sid_to_changeset(changeset) do
-    hub_sid = :crypto.strong_rand_bytes(16)
-                 |> Base.encode32
-                 |> String.downcase
-                 |> String.slice(0, 7)
+    hub_sid =
+      16
```
Comment (InfiniteLee) [inline lib/ret/hub.ex]: seems like this should be defined as a constant somewhere

File: priv/repo/migrations/20180408231444_create_session_stats_table.exs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      add(:entered_event_payload, :jsonb)
+      add(:entered_event_received_at, :utc_datetime)
+    end
+
+    for year <- 2018..@max_year,
+        month <- 0..11 do
+      with end_month <- rem(month + 1, 12),
+           end_year <- if(month == 11, do: year + 1, else: year) do
+        execute(
+          "create table ret0.session_stats_y#{year}_m#{month + 1} partition of session_stats
```
Comment (InfiniteLee) [inline priv/repo/migrations/20180408231444_create_session_stats_table.exs]: wow.

File: lib/ret_web/channels/hub_channel.ex [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+         response <- RetWeb.Api.V1.HubView.render("show.json", %{hub: hub}) do
+      existing_stat_count =
+        socket
+        |> SessionStat.stat_query_for_socket()
+        |> Repo.all()
+        |> length
+
+      unless existing_stat_count > 0 do
+        with session_id <- socket.assigns.session_id,
+             started_at <- socket.assigns.started_at,
```
Comment (InfiniteLee) [inline lib/ret_web/channels/hub_channel.ex]: identation seems off here

File: lib/ret_web/channels/hub_channel.ex [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    end
+  end
+
+  defp join_with_hub(socket, nil) do
+    {:error, %{message: "No such Hub"}}
+  end
+
+  defp handle_entered_event(socket, payload) do
+    with stat_attributes <- [
+           entered_event_payload: payload,
```
Comment (InfiniteLee) [inline lib/ret_web/channels/hub_channel.ex]: Don't know how much it matters, but also indentation here

File: lib/ret_web/channels/hub_channel.ex [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+         ] do
+      socket
+      |> SessionStat.stat_query_for_socket()
+      |> Repo.update_all(set: stat_attributes)
+    end
+  end
+
+  defp handle_max_occupant_update(socket, occupant_count) do
+    with hub <- Repo.get_by(Hub, hub_sid: socket.assigns.hub_sid),
+         max_occupant_count <- max(occupant_count, hub.max_occupant_count) do
```
Comment (InfiniteLee) [inline lib/ret_web/channels/hub_channel.ex]: and here

File: lib/ret_web/channels/hub_channel.ex [hunk 7]
```diff
@@ -0,0 +1,86 @@
+defmodule RetWeb.HubChannel do
+  @moduledoc "Ret Web Channel for Hubs"
+
+  use RetWeb, :channel
+
+  alias Ret.{Hub, Repo, SessionStat}
+
+  def join("hub:" <> hub_sid, _payload, socket) do
+    hub = Repo.get_by(Hub, hub_sid: hub_sid)
+    socket |> join_with_hub(hub)
```
Comment (InfiniteLee) [inline lib/ret_web/channels/hub_channel.ex]: might be better as:
`Repo.get_by(Hub, hub_sid: hub_sid) |> join_with_hub(socket)`
(and reorder the args in join_with_hub)

File: lib/ret_web/channels/hub_channel.ex [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+             changeset <- %SessionStat{} |> SessionStat.changeset(stat_attrs) do
+          Repo.insert(changeset)
+        end
+      end
+
+      {:ok, response, socket}
+    end
+  end
+
+  defp join_with_hub(socket, nil) do
```
Comment (InfiniteLee) [inline lib/ret_web/channels/hub_channel.ex]: `join_with_hub(_, nil)` or `join_with_hub(nil, _)` if you take my other suggestion above

File: priv/repo/migrations/20180408231444_create_session_stats_table.exs (hunk 3 above)
Comment (gfodor) [reply priv/repo/migrations/20180408231444_create_session_stats_table.exs]: postgresql 10 partitioning, w00t

**Pull request merged**